crates/red_knot_python_semantic/src/semantic_index/use_def/symbol_state.rs

```diff
@@ -310,7 +310,7 @@ impl SymbolState {
                 visibility_constraints: VisibilityConstraintPerBinding::default(),
             },
             declarations: SymbolDeclarations {
-                live_declarations: self.declarations.live_declarations.clone(),
+                live_declarations: Declarations::default(),
```

Comment:
Could we initialize all vecs here with their existing capacity or even use the max capacity between `a` and `b`? It seems to me that we always add at least all existing entries.

Reply:
Note that `live_declarations` is a `BitSet`, not a vector. Or do you think it would make sense to add something like a capacity parameter for large `BitSet`s that have overflowed onto the heap?

Reply:
Isn't the bitset dynamically sized? I'd suggest to have a `BitSet::with_capacity` so that it can allocate the right inner vector. We could consider doing the same for all other fields. E.g. `visibility_constraint`'s length is at least as large as the max of `a`'s and `b`'s visibility constraints (if I understand the logic correctly)

Reply:
> Isn't the bitset dynamically sized?

It is, I was mostly just responding to the "initialize all vecs here".

For small sizes, the `BitSet` is stack-allocated with a fixed capacity, so as long as we're below that threshold, reserving wouldn't help. But in general, adding reserve/with_capacity calls here sounds like a reasonable thing to try!